### PR TITLE
Support orthogonal embedding initialization

### LINKED ISSUE #5485: Orthogonal initializer dimensions error
Orthogonal Initializer throws an error whenever the first dimension is larger than the second. This could be fixed by transposing the array rather than throwing an error.

e.g. 1. ` dense = L.Linear(64,128, initialW=initializers.Orthogonal())`
image

e.g. 2 `initializers.generate_array(initializers.Orthogonal(), (20,10), numpy, 'f')`
image

System Info:
Chainer: 5.0.0rc1
NumPy: 1.14.2
CuPy:
  CuPy Version          : 5.0.0rc1
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 8000
  CUDA Driver Version   : 9020
  CUDA Runtime Version  : 8000
  cuDNN Build Version   : 7102
  cuDNN Version         : 7102
  NCCL Build Version    : 2213

### PR DESCRIPTION
**Pull request by toslunar**
Close #5485.

Merge #5615 first.

### CODE REVIEW

**Comment by hvy** [comment]
@beam2d Could you take a look at this one as well?

File: chainer/initializers/orthogonal.py [hunk 1]
```diff
@@ -53,14 +72,22 @@ def __call__(self, array):
...
-                                 ' # of vectors ({}) is larger than'
-                                 ' that of dimensions ({})'.format(
-                                     flat_shape[0], flat_shape[1]))
-            a = numpy.random.normal(size=flat_shape)
+            out_dim = len(array)
+            in_dim = utils.size_of_shape(array.shape[1:])
+            if (in_dim > out_dim and self._checks[0]) or (
+                    in_dim < out_dim and self._checks[1]):
+                raise ValueError(
+                    'Cannot make orthogonal {}.'
```
Comment (beam2d) [inline chainer/initializers/orthogonal.py]: Note: "auto" will not be used here, so the message will not corrupt.

**Comment by beam2d** [comment]
@toslunar Could you resolve the conflict? (sorry, it's due to my late review...)

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by beam2d** [comment]
LGTM. Jenkins failure is irrelevant.

**Pull request merged**